assets/js/util/i18n.js

```diff
@@ -33,28 +33,51 @@ import { __, sprintf } from '@wordpress/i18n';
  * For example, passing 65 returns '1m 5s'.
  *
  * @since 1.0.0
+ * @since n.e.x.t Refactored and renamed to improve localization.
+ * @private
  *
- * @param {number} seconds The number of seconds.
- * @return {string} Human readable string indicating time elapsed.
- *
+ * @param {number}                     seconds   The number of seconds.
+ * @param {(Intl.NumberFormatOptions)} [options] Optional formatting options.
+ * @return {string}   		Human readable string indicating time elapsed.
  */
-export const prepareSecondsForDisplay = ( seconds ) => {
+const durationFormat = ( seconds, options = { unitDisplay: 'short' } ) => {
 	seconds = parseInt( seconds, 10 );
 
-	if ( isNaN( seconds ) || 0 === seconds ) {
-		return '0.0s';
+	if ( Number.isNaN( seconds ) ) {
+		seconds = 0;
 	}
-	const results = {};
-	results.hours = Math.floor( seconds / 60 / 60 );
-	results.minutes = Math.floor( ( seconds / 60 ) % 60 );
-	results.seconds = Math.floor( seconds % 60 );
 
-	const returnString =
-		( results.hours ? results.hours + 'h ' : '' ) +
-		( results.minutes ? results.minutes + 'm ' : '' ) +
-		( results.seconds ? results.seconds + 's ' : '' );
+	let hours = Math.floor( seconds / 60 / 60 ) || '';
+	let minutes = Math.floor( ( seconds / 60 ) % 60 ) || '';
+
+	seconds = Math.floor( seconds % 60 );
+
+	if ( hours ) {
+		hours = numberFormat( hours, {
+			style: 'unit',
+			unit: 'hour',
+			...options,
+		} );
+	}
 
-	return returnString.trim();
+	if ( minutes ) {
+		minutes = numberFormat( minutes, {
+			style: 'unit',
+			unit: 'minute',
+			...options,
+		} );
+	}
+
+	seconds = numberFormat( seconds, {
+		style: 'unit',
+		unit: 'second',
+		...options,
+	} );
+
+	/* translators: 1: hours, 2: minutes, 3: seconds */
+	const formattedString = sprintf( _x( '%1$s %2$s %3$s', 'duration of time: hh mm ss', 'google-site-kit' ), hours, minutes, seconds );
```

Comment:
Looking at this again and thinking longer term, it seems reasonable that we might extend this to include larger units of time in the future (e.g. days, weeks...) at which point adding these placeholders would be a bit awkward.

Let's reverse the order so that `%1$s` refers to seconds, `%2$s` to minutes, and `%3$s` to hours so that the number of the placeholder increases with the unit of time.

Reply:
Good idea